addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -689,7 +690,38 @@ const MobileYoutubeAutoplayMixin = {
     },
 };
 
-registry.mediaVideo = publicWidget.Widget.extend(MobileYoutubeAutoplayMixin, {
+const videoNoCookieMixin = {
+    supportedDomains: [
+        "youtu.be", "youtube.com", "youtube-nocookie.com",
+        "instagram.com",
+        "player.vimeo.com", "vimeo.com",
+        "dailymotion.com",
+        "player.youku.com", "youku.com",
+    ],
+    /**
+     * @private
+     * @param {string} link
+     * @returns {string}
+     */
+    _getPrivacyFriendlyVideoLink(link) {
+        const linkUrl = new URL(link.startsWith("//") ? `https:${link}` : link);
+        if (this.supportedDomains.includes(linkUrl.host.replace(/^www\./, ""))) {
+            if (linkUrl.host.includes("vimeo")) {
+                // Returns a link suitable for embed videos when the optional
+                // cookies have not been accepted by the user.
+                linkUrl.searchParams.set("dnt", "1");
+                return linkUrl.href.replace(linkUrl.protocol, "");
+            }
+
+            return "about:blank";
+        }
+        // Due to video validations, we should not end here, but if it happens,
+        // consider it is privacy-friendly.
+        return link;
+    },
+};
+
+registry.mediaVideo = publicWidget.Widget.extend(MobileYoutubeAutoplayMixin, videoNoCookieMixin, {
```

Comment:
is this code just to use the vimeo cookie friendly url when possible?
If yes, it's not worth the code. If we had multiple main platforms benefiting from this code, why not, but here the technical debt seems to much

Reply:
I get it, but at the same time as someone who never accepts cookies, I like that there is at least one built-in way to set videos that are not blocked if optional cookies are not allowed (other than going through the embed code snippet with some other host).
For what it's worth, I also like the idea of rewarding services that give privacy-focused APIs...
(I started writing this function hoping there would be more than one platform to offer such API.)

Reply:
> I like that there is at least one built-in way to set videos that are not blocked if optional cookies are not allowed

For v18, we will have the video upload feature, which will solve this "issue" :+1:
Since there is only Vimeo, let's not add all this code for this, the technical debt is not good here IMHO.

Actually, why don't we force `dnt` when user insert a vimeo link? (would be unrelated to this PR)

Reply:
Yes, I know you're right, I just wanted to give my view on the matter. :smile: 

> Actually, why don't we force dnt when user insert a vimeo link?
I certainly won't be against it :)

Reply:
> Yes, I know you're right

Not especially, it's really just a matter of preference and past experience knowing that this code will eventually bug, there is nothing wrong apart from that in the current code, it just ends up in my "KISS" mindset here.

I created a task to investigate the forced `dnt` :+1: 
From what I understand, there is things that doesn't work with it like if you have multiple autoplay video on a page, they all start instead of only one, or the fact that if you have multiple video, when you play one it pauses others. Does not seems to really be an issue to me tho.

Reply:
https://www.odoo.com/web#menu_id=4720&cids=1&action=333&active_id=974&model=project.task&view_type=form&id=3724858